src/app/projects/pages/autofill-project-form/autofill-project-form.component.ts

```diff
@@ -140,8 +159,57 @@ export class AutofillProjectFormComponent implements OnInit {
     this.router.navigate(['/projects', 'register'], {queryParams: params});
   }
 
-  openDialog() {
-    const dialogRef = this.dialog.open(GeoAccessionDialogComponent);
-    dialogRef.disableClose = true;
+  onDoiExistence($event: string) {
+    this.hasDoi = $event == 'Yes' ? true : false;
+    this.hasGeo = this.hasDoi && (this.hasGeo != undefined || this.hasGeo) ? false : this.hasGeo;
+  }
+
+  onGEOAccessionExistence($event: string) {
+    this.hasGeo = $event == 'Yes' ? true : false;
+    this.hasDoi = this.hasGeo && (this.hasDoi != undefined || this.hasDoi) ? false : this.hasDoi;
+  }
+
+  private onUniqueGeoAccession(geoAccession) {
+    this.loaderService.display(true, 'This may take a moment. Please wait...');
+    this.brokerService.importProjectUsingGeo(geoAccession).pipe(
+      catchError(err => {
+        this.loaderService.display(true, `Sorry, we are unable to import the project yet due to error: [${err.message}]. You can still get a spreadsheet to import the project later.
+           We are now generating the spreadsheet, please wait this may take a moment...`);
+        return this.brokerService.downloadSpreadsheetUsingGeo(geoAccession);
+      })
+    ).subscribe(response => {
+        const projectUuid = response['project_uuid'];
+        if (projectUuid) {
+          this.onSuccessfulProjectImportFromGeo(geoAccession, projectUuid);
+        } else {
+          this.onSuccessfulSpreadsheetDownloadFromGeo(response, geoAccession);
+        }
+        this.loaderService.hide();
+      },
+      error => {
+        this.alertService.error('Unable to import the project or download spreadsheet', error.message);
+        this.loaderService.hide();
+      })
+  }
+
+  private onSuccessfulSpreadsheetDownloadFromGeo(response, geoAccession) {
```

Comment:
Yeah there's a real mix of GEO/general DOI stuff here. I think it needs to be broken up